src/Framework/PaymentGateways/Commands/PaymentComplete.php

```diff
@@ -5,21 +5,4 @@
 /***
  * @unreleased
  */
-class PaymentComplete implements GatewayCommand {
-    /**
-     * The Gateway Transaction / Charge Record ID
-     *
-     * @var string
-     */
-    public $gatewayTransactionId;
-
-    /**
-     * @unreleased
-     *
-     * @param  string  $gatewayTransactionId
-     */
-    public function __construct($gatewayTransactionId)
-    {
-        $this->gatewayTransactionId = $gatewayTransactionId;
-    }
-}
\ No newline at end of file
+class PaymentComplete extends PaymentCommand {}
```

Comment:
We'll need to make it clear that `$gatewayTransactionId` is _not_ optional for this command.  So let's overload the parent construct here to make it required.

Reply:
The transaction ID might be optional, though. In the case of BitPay we actually have to set the transaction ID up front, since we are dealing with invoices, which are then paid.

Reply:
Like the `withNotes()` method, we could add a `withTransactionID()` method so that setting the transaction ID is declarative.

Reply:
okay let's just keep it optional for now.

Reply:
See 11a2835.